### PR TITLE
Waiting for current leader topic partitions in taskConfigs() instead of start()

### PR DESCRIPTION
**Pull request by wy96f**
In the case of transient failure, getting current leader topic partitions in `KafkaSourceConnector::start` will timeout resulting in source connector failed and we have to manually restart connector until we have the resources.
We can lazily retrieve leader topic partitions and wait for them in `KafkaSourceConnector::taskConfigs`. If  failed, kafka connect framework will help us to retry the process, see https://github.com/apache/kafka/blob/6b6b36cec043eefc830c5209d5a18bbcb47a31db/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java#L1238

### CODE REVIEW

**Comment by rhysmccaig** [comment]
@wy96f thanks for your contribution. This looks good, I will try to review in detail and merge this week.

File: src/main/java/com/comcast/kafka/connect/kafka/PartitionMonitor.java [hunk 1]
```diff
@@ -149,11 +149,27 @@ private boolean matchedTopicFilter(String topic) {
     return topicWhitelistPattern.matcher(topic).matches();
   }
 
-  private synchronized void setCurrentLeaderTopicPartitions(Set<LeaderTopicPartition> leaderTopicPartitions) {
-    currentLeaderTopicPartitions = leaderTopicPartitions;
+  private void setCurrentLeaderTopicPartitions(Set<LeaderTopicPartition> leaderTopicPartitions) {
+    if (currentLeaderTopicPartitions == null) {
+      currentLeaderTopicPartitions = leaderTopicPartitions;
```
[the remark this answers is not in the document]
Comment (wy96f) [reply src/main/java/com/comcast/kafka/connect/kafka/PartitionMonitor.java]: Changed as requested :)

**Comment by rhysmccaig** [comment]
@wy96f I've merged this but I think there is additional work to be done to get the behavior you desired:

See: https://github.com/Comcast/MirrorTool-for-Kafka-Connect/blob/49ba1058145e5a655b9039e470cc3c8041982e07/src/main/java/com/comcast/kafka/connect/kafka/KafkaSourceConnector.java#L50

The source connector starts the partition monitor here, but this method blocks and waits until the initial list is fetched here, *before* starting the partition monitor thread:

https://github.com/Comcast/MirrorTool-for-Kafka-Connect/blob/49ba1058145e5a655b9039e470cc3c8041982e07/src/main/java/com/comcast/kafka/connect/kafka/PartitionMonitor.java#L130

I think the fix for this is simply to remove the above line, and schedule the executor to start the polling thread immediately, rather than have an initial delay of `topicPollIntervalMs` in the line below. 

https://github.com/Comcast/MirrorTool-for-Kafka-Connect/blob/49ba1058145e5a655b9039e470cc3c8041982e07/src/main/java/com/comcast/kafka/connect/kafka/PartitionMonitor.java#L144

You are welcome to send a new PR for this, otherwise I can take a look at it over the weekend when I have a little more time to test.

**Comment by wy96f** [comment]
@rhysmccaig Sent a new pr #23 to address this.

**Pull request merged**